{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/validation/notebooks/ModelValidation.py

```diff
@@ -172,15 +172,38 @@ def get_targets_from_recipe():
 
 # COMMAND ----------
 
-eval_result = None
-err = None
+# helper methods
+def get_run_link(run_info):
+    return "[Run](#mlflow/experiments/{0}/runs/{1})".format(
+        run_info.experiment_id, run_info.run_id
+    )
 
 
-def log_to_model_description(run, success):
-    run_info = run.info
-    run_link = "[Run](#mlflow/experiments/{0}/runs/{1})".format(
-        run_info.experiment_id, run_info.run_id
+def get_training_run(model_name, model_version):
+    model_versions = client.search_model_versions(f"name='{model_name}'")
```

Comment:
Do we need to search model versions here? Why not look up the model version directly?

Reply:
I investigated it for some time. This is the only way that I could find. 

The purpose is to get the training run info. We put its url in the validation run description and used its name as prefix.
However the mlflow recipe doesn't expose the training run information. We have to first get the registered model to find the run.
`Model.load("models:/prod-mlops-azure-cuj-model/83")` contains model metadata but doesn't have the run info.
```
databricks_runtime: 12.1.x-cpu-ml-scala2.12
flavors:
  python_function:
    artifacts:
      model_path:
        path: artifacts/sk_model
        uri: /root/.mlflow/recipes/77789c8c041dff48603f23e948b964c1e54512b9c0da38bf58dfba7bf19b9a1a/steps/train/outputs/sk_model
    cloudpickle_version: 2.0.0
    code: code
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.pyfunc.model
    python_model: python_model.pkl
    python_version: 3.9.5
  sklearn:
    code: code
    pickled_model: artifacts/sk_model/model.pkl
    serialization_format: cloudpickle
    sklearn_version: 1.2.1
mlflow_version: 2.1.1
model_uuid: 2ab6371e9fa146c6bdb603e23633ce76
signature:
  inputs: '[{"name": "tpep_pickup_datetime", "type": "datetime"}, {"name": "tpep_dropoff_datetime",
    "type": "datetime"}, {"name": "trip_distance", "type": "double"}, {"name": "pickup_zip",
    "type": "integer"}, {"name": "dropoff_zip", "type": "integer"}]'
  outputs: '[{"type": "tensor", "tensor-spec": {"dtype": "float32", "shape": [-1]}}]'
utc_time_created: '2023-02-20 09:11:54.732645'
```

Reply:
Right, what I mean is can you just look up the model version directly and avoid iterating over them?

https://mlflow.org/docs/latest/python_api/mlflow.client.html#mlflow.client.MlflowClient.get_model_version

Reply:
Tried and it works!

Reply:
There's exception if the version doesn't exist. Now I don't need to handle the error case.

Reply:
Yeah, but we are doing some exception handling anyways now (training run not found). It's not that big of a change.

There could be models with tons of model versions (thousands) - it's wasteful to fetch all and iterate on the client IMHO.